corehq/apps/users/static/users/js/roles.js

```diff
@@ -131,14 +132,14 @@ hqDefine('users/js/roles',[
                         };
                     }),
                 };
-
                 data.manageRoleAssignments = {
                     all: data.is_non_admin_editable,
                     specific: ko.utils.arrayMap(o.nonAdminRoles, function (role) {
                         return {
                             path: role._id,
                             name: role.name,
-                            value: data.assignable_by.indexOf(role._id) !== -1,
+                            value: ko.observable(data.assignable_by.indexOf(role._id) !== -1),
+                            access_all_locations: role.permissions.access_all_locations,
```

Comment:
This is an observable, right?

Reply:
huh, I see. I had thought it would need to be in order for the new markup to appear/disappear as needed when someone checks/unchecks full org access. You've confirmed that works, right?

Reply:
The js already rewraps the saved role with the data sent back from the server. I think you'd need to rewrap the other roles and/or recreate the `RolesViewModel` based on the new data.

That said, I checked out the branch and played around with the interaction, and I think it's fine to call that out of scope.

Reply:
I wouldn't think you'd need to add any data - each role has the proper info about the other roles on page load, that data just isn't getting refreshed properly somewhere.

But yeah, out of scope is fine too.